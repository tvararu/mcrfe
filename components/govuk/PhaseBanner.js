import { useState, useEffect } from "react";

const spaToggleHref = spaMode =>
  `javascript:document.cookie="spaMode=${
    spaMode ? "false" : "true"
  }";location.reload();`;

export default () => {
  const [rendered, setRendered] = useState(false);
  useEffect(() => {
    setRendered(true);
  }, []);
  return (
    <div className="govuk-phase-banner">
      <p className="govuk-phase-banner__content">
        <strong className="govuk-tag govuk-phase-banner__content__tag ">
          Prototype
        </strong>
        <span className="govuk-phase-banner__text">
          This is a prototype.{" "}
          <a className="govuk-link" href={spaToggleHref(rendered)}>
            Switch {rendered ? "off" : "on"} experimental single-page mode.
          </a>
        </span>
      </p>
    </div>
  );
};
