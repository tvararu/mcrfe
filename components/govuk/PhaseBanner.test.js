import renderer from "react-test-renderer";
import { render, fireEvent, cleanup } from "@testing-library/react";
import PhaseBanner from "./PhaseBanner";

describe("PhaseBanner", () => {
  it("renders correctly", () => {
    const component = renderer.create(<PhaseBanner />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("behaviour", () => {
    afterEach(cleanup);

    it("renders correctly", () => {
      const { container } = render(<PhaseBanner />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
