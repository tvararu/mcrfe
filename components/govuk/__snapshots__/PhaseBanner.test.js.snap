// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseBanner behaviour renders correctly 1`] = `
<div
  class="govuk-phase-banner"
>
  <p
    class="govuk-phase-banner__content"
  >
    <strong
      class="govuk-tag govuk-phase-banner__content__tag "
    >
      Prototype
    </strong>
    <span
      class="govuk-phase-banner__text"
    >
      This is a prototype.
       
      <a
        class="govuk-link"
        data-qa="spa-switch"
        href="javascript:document.cookie=\\"spaMode=false\\";location.reload();"
      >
        Switch 
        off
         experimental single-page mode.
      </a>
    </span>
  </p>
</div>
`;

exports[`PhaseBanner renders correctly 1`] = `
<div
  className="govuk-phase-banner"
>
  <p
    className="govuk-phase-banner__content"
  >
    <strong
      className="govuk-tag govuk-phase-banner__content__tag "
    >
      Prototype
    </strong>
    <span
      className="govuk-phase-banner__text"
    >
      This is a prototype.
       
      <a
        className="govuk-link"
        data-qa="spa-switch"
        href="javascript:document.cookie=\\"spaMode=true\\";location.reload();"
      >
        Switch 
        on
         experimental single-page mode.
      </a>
    </span>
  </p>
</div>
`;
